docs/software/uenv/deploy.md

```diff
@@ -226,10 +227,45 @@ Permissions to access restricted resources is set on a per-pipeline basis
 | `vasp` | `vasp6`, `cscs-uenv-admin` | `uenv-sources/vasp` | VASP requires a paid license to access source | Simon Frasch |
 | `vmd` | `uenv-sources-csstaff` | `uenv-sources/vmd` | VMD requires an account to download the source code | Alberto Invernizzi |
 
-[](){#ref-uenv-removal}
-## Deprecation and removal of uenv
+[](){#ref-uenv-lifecycle}
+## uenv lifecycle
 
-!!! todo "Finalize and document the deprecation process"
+Scientific applications and tools have different release cycles (e.g. annual, quarterly or irregular), and the communities that use them have different expectations (e.g. ML users expect tools released in the last 3 months, while some scientific communities value support for old versions of software).
+For this reason, there is not a universal release and deprecation schedule for supported applications and programming environments - how often to release, how long to provide support, and when to remove old versions is up to the maintainer of the uenv, based on their user's requirements and the overheads of maintaining old versions.
 
+While the frequency of updates and deprecation policy is defined by the uenv maintainer, the following release cycle with tag names should be followed when possible:
+
+- [optional] tag pre-releases with `:rc1`, `:rc2`, etc
+    - uenv tagged as release candidates are intended for early testing only, and can be removed at any point.
+    - e.g. `prgenv-gnu/25.6:rc1` is released for users to do early testing and give feedback.
+- Tag `:v1` for the first official release.
+- [optional] release updated versions of uenv with tags ()`:v2`, `:v3`, ...) with patches and fixes.
+- Remove old versions according to a deprecation policy.
+
+A uenv's release and support policy should be part of the uenv's documentation under the "Versioning" section, see [prgenv-gnu][ref-uenv-prgenv-gnu-versioning] for example.
+
+The versioning section should provide the following information:
+
+* the release schedule;
+* the support, update and deprecation policy for the uenv;
+* details about the currently deployed versions, changelogs and relevant information how to upgrade if appropriate.
+
+!!! example "Example deprecation policy"
```

Comment:
```suggestion
??? example "Example deprecation policy"
```

Collapsed? Useful mostly for us, while users will have moved to the `uenv`-specific documentation to see the actual policy at this point.